hazelcast/src/main/java/com/hazelcast/spi/impl/eventservice/impl/EventServiceImpl.java

```diff
@@ -441,16 +492,10 @@ public void executeEventCallback(Runnable callback) {
 
     @Override
     public void handle(Packet packet) {
-        try {
-            eventExecutor.execute(new RemoteEventProcessor(this, packet));
-        } catch (RejectedExecutionException e) {
-            rejectedCount.inc();
-
-            if (eventExecutor.isLive()) {
-                Connection conn = packet.getConn();
-                String endpoint = conn.getEndPoint() != null ? conn.getEndPoint().toString() : conn.toString();
-                logFailure("EventQueue overloaded! Failed to process event packet sent from: %s", endpoint);
-            }
+        EventEnvelope eventEnvelope = (EventEnvelope) nodeEngine.toObject(packet);
```

Comment:
I was meaning, `RemoteEventProcessor` is used to offload that de-serialization work to another thread than IO thread, now, with your change, it will be an extra work for IO thread (beware that method `handle(Packet packet)` is called by IO thread). And given that IO thread is a shared resource and it mostly only dispatches packets to handlers, seems better to revert your change for that.